qiskit_experiments/library/calibration/fine_amplitude.py

```diff
@@ -12,297 +12,144 @@
-                )
-
-        # Prepare the circuits.
-        repetitions = self.experiment_options.get("repetitions")
-
-        circuits = []
-
-        if self.experiment_options.add_xp_circuit:
-            # Note that the rotation error in this xval will be overweighted when calibrating xp
-            # because it will be treated as a half pulse instead of a full pulse. However, since
-            # the qubit population is first-order insensitive to rotation errors for an xp pulse
-            # this point won't contribute much to inferring the angle error.
-            angle_per_gate = self.analysis_options.get("angle_per_gate", None)
-            phase_offset = self.analysis_options.get("phase_offset")
-
-            if angle_per_gate is None:
-                raise CalibrationError(
-                    f"Unknown angle_per_gate for {self.__class__.__name__}. "
-                    "Please set it in the analysis options."
-                )
-
-            circuit = QuantumCircuit(1)
-            circuit.x(0)
-            circuit.measure_all()
-
-            circuit.metadata = {
-                "experiment_type": self._type,
-                "qubits": self.physical_qubits,
-                "xval": (np.pi - phase_offset) / angle_per_gate,
-                "unit": "gate number",
-            }
+        options.result_index = -1
+        options.target_angle = np.pi
+        options.group = "default"
 
-            circuits.append(circuit)
+        return options
 
-        for repetition in repetitions:
-            circuit = self._pre_circuit()
+    def update_calibrations(self, experiment_data: ExperimentData):
+        r"""Update the amplitude of the pulse in the calibrations.
 
-            for _ in range(repetition):
-                circuit.append(gate, (0,))
+        The update rule of this experiment is
 
-            circuit.measure_all()
+        .. math::
 
-            if schedule is not None:
-                circuit.add_calibration(gate, self.physical_qubits, schedule, params=[])
+            A \to A \frac{\theta_\text{target}}{\theta_\text{target} + {\rm d}\theta}
 
-            circuit.metadata = {
-                "experiment_type": self._type,
-                "qubits": self.physical_qubits,
-                "xval": repetition,
-                "unit": "gate number",
-            }
+        Where :math:`A` is the amplitude of the pulse before the update.
 
-            circuits.append(circuit)
+        Args:
+            experiment_data: The experiment data from which to extract the measured over/under
+                rotation used to adjust the amplitude.
+        """
+        result_index = self.experiment_options.result_index
+        group = self.experiment_options.group
+        target_angle = self.experiment_options.target_angle
 
-        return circuits
+        d_theta = BaseUpdater.get_value(experiment_data, "d_theta", result_index)
 
+        prev_amp = self._cals.get_parameter_value(
```

Comment:
I wonder if we could store `prev_amp` in the metadata and/or store the new amp as a result? I am thinking of the case when a user is looking back at this calibration. It will be harder to look up the previous amplitude in Calibrations to figure out what amplitude the calibration found then after Calibrations has possibly been updated again -- reloading the experiment and rerunning the analysis would possibly work from a different `prev_amp` value. One could look at the Calibrations history possibly but it seems nice to make the new amplitude result accessible from the experiment as well.